engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -550,7 +550,7 @@ namespace dmGameObject
      * @name go.get
      * @param url [type:string|hash|url] url of the game object or component having the property
      * @param property [type:string|hash] id of the property to retrieve
-     * @param options [type:table] (optional) options table
+     * @param [options] [type:table] optional options table
```

Comment:
Question is, would this actually work with our current scripts?
These changes aren't tested, right?

Reply:
Let me test this!

Reply:
Generated to this in `go_doc.script_api`:

```
  - desc: gets a named property of the specified game object or component
    name: get
    parameters:
    - desc: url of the game object or component having the property
      name: url
      type:
      - string
      - hash
      - url
    - desc: id of the property to retrieve
      name: property
      type:
      - string
      - hash
    - desc: 'optional options table

        - index [type:integer] index into array property (1 based)

        - key [type:hash] name of internal property'
      name: options
      optional: true
      type: table
    returns:
    - desc: the value of the specified property
      type: any
    type: function
```

BUT the `go_doc.json` doesn't look quite right:


```
    {
      "type": "FUNCTION",
      "name": "go.get",
      "brief": "gets a named property of the specified game object or component",
      "description": "gets a named property of the specified game object or component",
      "returnvalues": [
        {
          "name": "value",
          "doc": "the value of the specified property",
          "types": [
            "any"
          ]
        }
      ],
      "parameters": [
        {
          "name": "url",
          "doc": "url of the game object or component having the property",
          "types": [
            "string",
            "hash",
            "url"
          ]
        },
        {
          "name": "property",
          "doc": "id of the property to retrieve",
          "types": [
            "string",
            "hash"
          ]
        },
        {
          "name": "[options]",
          "doc": "optional options table\n- index <span class=\"type\">integer</span> index into array property (1 based)\n- key <span class=\"type\">hash</span> name of internal property",
          "types": [
            "table"
          ]
        }
      ],
      "examples": "Get a named property\n<div class=\"codehilite\"><pre><span></span><code><span class=\"kr\">function</span> <span class=\"nf\">init</span><span class=\"p\">(</span><span class=\"n\">self</span><span class=\"p\">)</span>\n    <span class=\"c1\">-- get the resource of a certain gui font</span>\n    <span class=\"kd\">local</span> <span class=\"n\">font_hash</span> <span class=\"o\">=</span> <span class=\"n\">go</span><span class=\"p\">.</span><span class=\"n\">get</span><span class=\"p\">(</span><span class=\"s2\">&quot;#gui&quot;</span><span class=\"p\">,</span> <span class=\"s2\">&quot;fonts&quot;</span><span class=\"p\">,</span> <span class=\"p\">{</span><span class=\"n\">key</span> <span class=\"o\">=</span> <span class=\"s2\">&quot;system_font_BIG&quot;</span><span class=\"p\">})</span>\n<span class=\"kr\">end</span>\n</code></pre></div>",
      "replaces": "",
      "error": "",
      "tparams": [],
      "members": [],
      "notes": []
    },
```

I was expecting the `options` table to be set as optional, like in the script_api. Perhaps this was not how it worked before either?

Reply:
> Perhaps this was not how it worked before either?

Correct, that part wasn't working before either.

Reply:
We are checking for optional arguments in the generator:

https://github.com/defold/defold/blob/dev/engine/docs/src/script_doc.py#L330-L335